src/Xamarin.Android.Build.Tasks/Properties/Resources.resx

```diff
@@ -167,8 +167,8 @@ In this message, the phrase "should not be reached" means that this error messag
     <comment>The abbreviation "JDK" should not be translated.</comment>
   </data>
   <data name="XA0031" xml:space="preserve">
-    <value>Java SDK {0} or above is required when using $(TargetFrameworkVersion) {1}.</value>
-    <comment>The following are literal names and should not be translated: $(TargetFrameworkVersion)
+    <value>Java SDK {0} or above is required when using {1}.</value>
```

Comment:
Should we change this to this to say `Java SDK {0} or above is required.`? I think continuing to print a target framework version could be confusing?

Reply:
I was trying to use the exact same message in both old and new. Especially since it looks like we have it translated already...

And example of the message for legacy tasks in this PR will be:

    Java SDK 1.8.0 or above is required when using $(TargetFrameworkVersion) v10.0.

I moved the word `$(TargetFrameworkVersion)` to the legacy MSBuild task.

The new message will be:

    Java SDK 1.8.0 or above is required when using Xamarin.Android.Sdk.

`Xamarin.Android.Sdk` is kind of a placeholder. We should figure out what wording is ok for the new one, can we just put something in the `{1}` that will work? @brendanzagaeski what do you think?

Reply:
Maybe it could be:
```
Java SDK 1.8.0 or above is required when using .NET 5.
```

Reply:
If we want to go that route maybe we should do `... when using .NET 5 or newer.` or something like that, or we'll have to consider bumping this message regularly. I'm fine with `Xamarin.Android.Sdk` for now though.

Reply:
`... when using .NET 5 or newer` would just mean it needs to be translated.

Reply:
I think it'd be fine to add another string resource for the new tasks. It could use a property name like `XA0031_NET5`. That'll give flexibility to adjust the wording as the vocabulary around .NET 5 solidifies. For example, "when using .NET 5" might not be exactly true since a long-form project in a .NET 5 release of Xamarin.Android would still have the old behavior, if I'm understanding correctly?

An expansion on the original idea might be to include extra context about the term "Xamarin.Android.Sdk", like:

```
Java SDK 1.8.0 or above is required for projects that use `<Project Sdk="Xamarin.Android.Sdk">`.
```

The most solid related vocabulary term might be "[Sdk attribute](https://docs.microsoft.com/dotnet/core/tools/csproj#sdk-attribute)," but I have a hunch the example XML snippet might be more recognizable.

From what I understand, the individual strings are handled pretty quickly by the translation team, and I think they might even have tooling to reuse parts of translations that closely match existing strings. So I think it will be an OK trade-off to request a new translation for this special case.